assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/llava/llava_mlflow_wrapper.py

```diff
@@ -160,6 +160,16 @@ def predict(self, context: mlflow.pyfunc.PythonModelContext, input_data: pd.Data
             ).unsqueeze(0).cuda()
             stopping_criteria = KeywordsStoppingCriteria([self._stop_str], self._tokenizer, input_ids)
 
+            # For small models on V100 machines, long prompts cause a GPU OOMs which the server does not recover from.
+            # To prevent this, we are using a length threshold that allows for a small number of question-answer pairs
+            # (e.g. 5-10) in each prompt.
+            if self._model_version in [self.LLAVA_MPT, self.LLAVA_7B]:
+                prompt_length = max([len(i) for i in input_ids])
+                if prompt_length > MAX_PROMPT_LENGTH:
+                    raise ValueError(
+                        f"Prompt too long: {prompt_length} tokens. Maximum allowed is {MAX_PROMPT_LENGTH}."
```

Comment:
The user doesn't really have any notion of "Prompt". Should we instead say "Dialogue" or "Conversation"?

Reply:
They do know what a prompt is. If not general knowledge, then they can read the model card. Speaking of which, turns out it **is** important to update the model card as per bug 2730582.